### PR TITLE
[fix] Prevent empty voucher inputs and empty invoices

### PR DESCRIPTION
**Pull request by ronniebeggs**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Added validation to prevent empty voucher inputs
- Restructured voucher amount validation to avoid extra database query
- Trigger navigation to the home page from the review and submit screen when there are zero vouchers in the invoice (prevents empty invoices)
- Autofill voucher amount text field to the max value for the given voucher range

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"
  
^Autofilled 10 dollar maxValue

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"
- Navigate to the confirm value screen and observe an auto-filled max value
- Observe errors triggered on empty amount inputs
- Navigate to the review and submit screen and delete all vouchers that you created
- Observe an empty invoice dialog that redirects you to the home page

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
[input field bug fix PR](https://github.com/calblueprint/vouchers4veggies/pull/70)

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."
🥦 CC: @sauhardjain

### CODE REVIEW

File: src/database/queries.ts [hunk 1]
```diff
@@ -171,54 +171,30 @@ export const getVoucher = async (serialNumber: number): Promise<Voucher> => {
 /**
  * Query the `vouchers` collection and check if a serialNumber is valid.
```
Comment (sauhardjain) [inline src/database/queries.ts]: Change comments to describe new query name & behavior.

**Review by sauhardjain — APPROVED** [review]
Looks good. One minor change in comments.

File: src/database/queries.ts [hunk 2]
```diff
@@ -171,54 +171,30 @@ export const getVoucher = async (serialNumber: number): Promise<Voucher> => {
...
-): Promise<boolean> => {
+): Promise<number> => {
   try {
     const docId = serialNumber.toString();
     const docRef = doc(db, 'vouchers', docId);
     // check that serialNumber exists
     const voucherRange = await getVoucherRange(serialNumber);
     if (voucherRange === null) {
-      return false;
+      return 0;
```
Comment (selene-huang) [inline src/database/queries.ts]: I think in general we should avoid using dirty return values as a way of indicating that the function has failed, because this is not necessarily intuitive and requires extensive documentation to note. Instead, I'd recommend either throwing an error here and catching it when you use `getMaxVoucherValue`, or you can see how error handling was done in `createVoucher` as an example.

File: src/screens/scanning/ReviewScreen.tsx [hunk 3]
```diff
@@ -99,7 +101,11 @@ export default function ReviewScreen({
   };
 
   const onDeleteHelper = () => {
+    const { size } = voucherMap;
     deleteVoucher(dispatch, focusedSerialNumber);
+    if (size <= 1) {
```
Comment (selene-huang) [inline src/screens/scanning/ReviewScreen.tsx]: I wonder if it's really best to lump this with onDelete functionality? It was pretty jarring to me when the dialog suddenly popped up after I deleted the voucher and forced me to return to the start. Personally I would've expected to see that kind of error dialog if I tried to submit the empty invoice, but not the moment the invoice becomes empty.

**Review by selene-huang — CHANGES_REQUESTED** [review]
Looks good overall, but I have some concerns about error handling in the queries and a suggestion for the current flow.

File: src/screens/scanning/ReviewScreen.tsx (hunk 3 above)
Comment (sauhardjain) [reply src/screens/scanning/ReviewScreen.tsx]: Yeah, I see your point here. @ronniebeggs let's display the error dialog on clicking the Submit button instead.

File: src/database/queries.ts (hunk 2 above)
Comment (sauhardjain) [reply src/database/queries.ts]: ++
